### ISSUE TITLE
Move Element-adapter-firestore to element-lib

### ISSUE THREAD

**Issue by gjgd** [decentralized-identity/element#97]
[this](https://github.com/decentralized-identity/element/blob/master/packages/element-api/src/services/element-adapter-firestore/index.js) should live [here](https://github.com/decentralized-identity/element/tree/master/packages/element-lib/src/adapters/database)

**Comment by OR13** [comment]
I'm not sure I agree.

The lib should implement only the adapters that are needed for web / nodejs out of the box. custom adapters should be built as seperate npm modules, like they are in pouchdb...

https://pouchdb.com/adapters.html

**Comment by gjgd** [comment]
I think it would be convenient to have a good choice of adapters in the element-lib

**Comment by OR13** [comment]
yes I agree, do we think that Firestore (a closed source no sql data base that is difficult to migrate away from) should be bundled with element lib? I'd prefer to see RXDB Adapter and a couchdb adapter that works in both the browser and the backend server.

I don't want to bundle anything with element that does not run in both he web and node javascript environment or that enables vendor lockin.
